apps/meteor/client/views/admin/settings/inputs/TimespanSettingInput.tsx

```diff
@@ -0,0 +1,122 @@
+import { Field, FieldLabel, FieldRow, InputBox, Select } from '@rocket.chat/fuselage';
+import { useTranslation } from '@rocket.chat/ui-contexts';
+import type { FormEventHandler, ReactElement } from 'react';
+import React, { useMemo, useState } from 'react';
+
+import { TIMEUNIT, timeUnitToMs, msToTimeUnit } from '../../../../lib/convertTimeUnit';
+import ResetSettingButton from '../ResetSettingButton';
+import type { SettingInputProps } from './types';
+
+type TimespanSettingInputProps = SettingInputProps<string, string | number> & {
+	value: string;
+};
+
+export const getHighestTimeUnit = (value: number): TIMEUNIT => {
+	const minutes = msToTimeUnit(TIMEUNIT.minutes, value);
+	if (minutes % 60 !== 0) {
+		return TIMEUNIT.minutes;
+	}
+
+	const hours = msToTimeUnit(TIMEUNIT.hours, value);
+	if (hours % 24 !== 0) {
+		return TIMEUNIT.hours;
+	}
+
+	return TIMEUNIT.days;
+};
```

Comment:
is this correct? the name of the function isn't very clear to me, but **my guess is** it's trying to see the longest the number goes. 

I see two issues here,

first is what if in L15, minutes is literally 61? 

Second, this assumes two types, ints and floats, and assumes all conversions up to `%` are in ints. That is not true.

msToTimeUnit can return with a decimal part while integer part is > 0 (or multiplier for the respective unit).

imagine passing a huge number (milliseconds) that is divisible by 1000 (for second), 60 (for minute), 60 (hour), **plus 1**. 3600001.

When converted to minutes using your function, you get 60.<something>. Pass through the first branch and you get a false positive. 60.001 (e.g.) % 60 != 0, yes, but that doesn't mean the highest unit for that number is minutes, it is actually hour.

Reply:
The input does not accept not-whole values, so you can’t put in `1,5 hours` or something similar. If you want to be more granular you should go to the previous time unit. This is more of a UX issue than implementation issue, if you still think this shouldn't be as it is, the design team should be included in the discussion.

IMO the way it is provides enough granularity without increasing the complexity of the UI too much. What you pointed out about 60 minutes being treated as an hour, and 61 minutes being treated as minutes is actually how I intended this to be. We do not save the selected time unit, just the time in milliseconds, so the purpose of the function is to get the highest time unit that is a whole number.

If you check [this line](https://github.com/RocketChat/Rocket.Chat/pull/32425/files#diff-18a2cccf8325476785643aba44e04dc1d395aa4eb9727f2b98e47b7e6d6566b6R18) of the unit tests you'll see that I tested the specific case you're pointing out.